src/app/shared/loader.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Subject } from 'rxjs';
+import { LoaderState } from '../core/loader-state.model';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoaderService {
+
+  private loaderState = new Subject<LoaderState>();
+
+  public getLoaderState() {
+    return this.loaderState;
+  }
+
+  public startLoading() {
+    this.loaderState.next({loading: true});
```

Comment:
Could be simplified to dispatch just boolean instead object

Reply:
Yes, you are right. Thank you.